### ISSUE TITLE
DB Button does not work if a different local shipname is set.

### ISSUE THREAD

**Issue by aBlueShadow** [daid/EmptyEpsilon#1080]
If the locale name of a ship is different from the general name, the DB button on science is not functioning, as no DB entry is found.

#898 has been proposed to fix this (at the price of reverting the old behaviour of showing the general names in the DB), however it conflicts now, and I am not familiar with the new DB structure in order to fix this.

To reproduce this issue without a localized po in place: Spawn a "Cruiser"*, scan it and try to use the DB button.

*Not the player ship, so spawn it in GM screen.

**Comment by czenker** [comment]
> #898 has been proposed to fix this (at the price of reverting the old behaviour of showing the general names in the DB), however it conflicts now, and I am not familiar with the new DB structure in order to fix this.

Not much has changed in the ScienceDatabase. Basically I just turned the datastructure from a tree to a referenced list. Most of the API has stayed the same. The line has just moved a little down: https://github.com/daid/EmptyEpsilon/blob/master/src/scienceDatabase.cpp#L336

With such a simple pull request it is probably easier to create a new pull request and close the old one.

**Comment by aBlueShadow** [comment]
@czenker I see, so I guess that part would really be simple.
However, I assume that the old request was not merged in the first play because of the trade-off of not having the localized name in the DB.
Originally, it was part of a bigger PR( #896), that would have taken care of it by introducing a seperate locale_name property into the DB. I am not really deep into c++ programming, and after the collision with your DB restructuring I must admit, the new structure looked quite intimidating at a first glance to me. I figured it would take quite some time to look into and ultimately lost sight with the topic.
If you say it is not as complicated as it looked to me back then to me, I should maybe take another look into it, though as I said my C++ knowledge is rather rudimentary.

**Comment by tdelc** [comment]
I test on my fork with this correction. it seems ok : 

![image](https://user-images.githubusercontent.com/13868149/98823077-ef329880-2431-11eb-8c52-46e387b9841a.png)

**Comment by aBlueShadow** [comment]
I tried this patch, and it works. However, when I tested it with mixed locale environment (server original and client localized, and vice versa) I experienced occasionally that the science db got doubled.
I could not reliably reproduce this issue, so I am not sure it it is caused by this patch at all.
@czenker what are your thoughts about it?
Otherwise I would like to include this patch until the next release

**Comment by aBlueShadow** [comment]
Looks like that duplication was not related to your patch , but that I actually experienced #1240, probably due to some reconnection during testing. After #1254 being merged, I can't replicate this behavior anymore, so the patch seems alright.

**Comment by muerteFR** [comment]
On the latest version today, this problem is still relevant,
 whatever the language: fr, cz, it. (daaa kept the original names) Should we then put the original names back or is a "patch" planned?
Thx

**Comment by aBlueShadow** [comment]
#2517 now fixed this
